Adds recent Jupyter Notebooks card to Central Dashboard

prodonjs opened the pull request:
closes #3120 

/area front-end
/area centraldashboard
/cc @avdaredevil 
/cc @ajayalfred

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/kubeflow/kubeflow/3377)
<!-- Reviewable:end -->

prodonjs commented:
It's starting to look like we have a real dashboard on our hands here! 

The image with these changes is pushed to https://kf-dev.endpoints.prodonjs-kubeflow-dev.cloud.goog/.

A couple of things to note:
- Jupyter servers are namespaced, so the card is not populated until a namespace is chosen.
- I use a generic material horizontal loading indicator when the requests to get Notebooks are pending. I don't think we have a spec for that so feel free to provide a better suggestion for giving some kind of activity indicator.
- Results are limited to the 5 most recently modified notebooks. That seems to keep the card height where we'd want it. But, since our API requests to the Jupyter servers return all results, another option would be to add a pagination control to the card's footer.

ajayalfred commented:
Looks hot!!!

![image](https://user-images.githubusercontent.com/2425179/58651108-489e1a80-82c5-11e9-8c9e-4676a7f49794.png)
Change the empty state message to "No Notebooks in namespace (name of namespace)"

The horizontal loader looks great. I noticed it's a few pixels below the pin line though. Can you make it appear just under the pinline? See spec below
![image](https://user-images.githubusercontent.com/2425179/58651589-83548280-82c6-11e9-87d3-c1c0dab99636.png)

Regarding no. of results – I think we can start with 5 most recent to begin with. I wouldn't want to add pagination in the dashboard since it'll feel heavy handed and the idea of a dashboard is to be a quick glance-and-bounce page. We could consider increasing the no. of results to 10 since the dashboard is pretty empty at the moment. That said, it's likely that the user will find their most recently used notebook in the first 5 results :)

prodonjs commented:
@ajayalfred Changes to the loading indicator and the column spacing are in place and should be live in https://kf-dev.endpoints.prodonjs-kubeflow-dev.cloud.goog/.

@avdaredevil The latest commit (bf1ac3b) fixes an issue I noticed whereby the profile image was trying to load from /undefined. It took a little digging but apparently the `iron-image` component explicitly checks for an [empty string](https://github.com/PolymerElements/iron-image/blob/master/iron-image.js#L318) rather than a falsy value so because the user.image property was undefined, it was trying to fetch it. There are other ways of addressing it but this seemed to be the simplest.

LMK if there's anything else.

prodonjs commented:
/hold

There's a CSS issue with the dashboard-view now that I removed width: 100%. The scrollbar won't be at the far-right once we have more vertical content. However, please review the other JS changes.

prodonjs commented:
/hold cancel
This should be ready for final review. @ajayalfred I addressed some of the points you made on issue #3236 with respect to moving the charts to the first column and the notebooks card to the second. I also forced the x-axis labels to remain horizontal and shrunk the secondary text on each card item so that it's consistent across the board.

Changes are pushed the cluster I have linked to previously. The middle column looks a little bare when you first load the page before selecting a namespace, but that will improve once we add the two Pipelines cards which I should have a PR out today for.

avdaredevil commented on components/centraldashboard/public/components/dashboard-view.css:
```diff
@@ -31,13 +22,32 @@
+.column {
+    grid-row: 1;
+    min-width: 320px;
+    max-width: 400px;
```
If we have this logic defined here, all cards should no longer have this logic, and instead should be:
```css
width: 100%
```

prodonjs replied:
Agreed. I think I took care of this already by removing the widths from the `--dashboard-card` mix-in.

avdaredevil replied:
Ah, I may have missed that. Feel free to resolve

avdaredevil commented on components/centraldashboard/public/components/dashboard-view.pug:
```diff
@@ -1,44 +1,47 @@
-                a.link(href$='[[item.link]]', tabindex='-1', target='_blank')
+        paper-card#Quick-Links(heading='Quick shortcuts')
+            template(is='dom-repeat', items='[[quickLinks]]')
+                iframe-link.link(role='link', href$="[[item.link]]")
```
Instead of providing role here, can we have the `role` defined from within the component, defaulted to `link`?

prodonjs replied:
Good call. For some reason the idea of setting the attribute in the component's setup eluded me. Thanks for the suggestion!

avdaredevil commented:
Hey @prodonjs made a quick change, lemme know your thoughts?

avdaredevil commented:
/hold
For review from @prodonjs 

/lgtm
/approve

prodonjs commented:
I like it!

/hold cancel

ajayalfred commented:
Loader position looks perfect! 
The blue and gray color don't appear to match the spec. Can you change the blue to #669DF6 and the gray container to #F1F3F4? Thanks!

prodonjs commented:
Yes, I will get those colors fixed on #3379 .
